components/integrations/bots/add_bot/add_bot.tsx

```diff
@@ -13,178 +13,201 @@ import {General} from 'mattermost-redux/constants';
 
 import BotDefaultIcon from 'images/bot_default_icon.png';
 
-import {browserHistory} from 'utils/browser_history';
 import BackstageHeader from 'components/backstage/components/backstage_header.jsx';
 import OverlayTrigger from 'components/overlay_trigger';
 import Tooltip from 'components/tooltip';
 import SpinnerButton from 'components/spinner_button';
 import FormattedMarkdownMessage from 'components/formatted_markdown_message.jsx';
 import FormError from 'components/form_error';
+
+import {browserHistory} from 'utils/browser_history';
 import {AcceptedProfileImageTypes, Constants, ValidationErrors} from 'utils/constants';
 import * as Utils from 'utils/utils';
 
+import {Team} from '@mattermost/types/teams';
+import {Bot, BotPatch} from '@mattermost/types/bots';
+import {UserProfile} from '@mattermost/types/users';
+import {ActionResult} from 'mattermost-redux/types/actions';
+
 const roleOptionSystemAdmin = 'System Admin';
 const roleOptionMember = 'Member';
 
-export default class AddBot extends React.PureComponent {
-    static propTypes = {
+export type Props = {
 
-        /**
-        *  Only used for routing since backstage is team based.
-        */
-        team: PropTypes.object.isRequired,
+    /**
+     *  Only used for routing since backstage is team based.
+     */
+    team: Team;
 
-        /**
-        *  Bot to edit (if editing)
-        */
-        bot: PropTypes.object,
+    /**
+     *  Bot to edit (if editing)
+     */
+    bot?: Bot;
+
+    /**
+     *  Bot user
+     */
+    user?: UserProfile;
+
+    /**
+     *  Roles of the bot to edit (if editing)
+     */
+    roles?: string;
+
+    /**
+     * Maximum upload file size (for bot account profile picture)
+     */
+    maxFileSize: number;
+
+    /**
+     * Editing user has the MANAGE_SYSTEM permission
+     */
+    editingUserHasManageSystem: boolean;
+
+    /**
+     * TODO
+     */
+    location: {search: string};
```

Comment:
There was no `location` prop in the old version. However, `location` is used in the index so I added it

Reply:
I think you can get this type from the `RouteComponentProps` type from `react-router`. It's a type that contains `location` and several other routing-related props that you don't need otherwise

Reply:
Thanks for the suggestion. Done